client/js/components/SelfGuidedItem/SelfGuidedItem.js

```diff
@@ -24,54 +24,66 @@ const SelfGuidedItem = ({selfGuidedItem, navigation}) => (
+                    <Text style={styles.infobox}>
+                      {selfGuidedItem.difficulty
+                        .toLowerCase()
+                        .charAt(0)
+                        .toUpperCase() +
+                        selfGuidedItem.difficulty.toLowerCase().slice(1)}
+                    </Text>
+                  </View>
+                  <View style={styles.flexColumn}>
+                    <Text style={styles.subTitles}>Distance</Text>
+                    <Text style={styles.infobox}>
+                      {selfGuidedItem.distance}
+                    </Text>
+                  </View>
+                  <View style={styles.flexColumn}>
+                    <Text style={styles.subTitles}>Duration</Text>
+                    <Text style={styles.infobox}>
+                      {selfGuidedItem.duration} hours
+                    </Text>
+                  </View>
                 </View>
-                <View style={styles.flexColumn}>
-                  <Text style={styles.subTitles}>Distance</Text>
-                  <Text style={styles.infobox}>{selfGuidedItem.distance}</Text>
-                </View>
-                <View style={styles.flexColumn}>
-                  <Text style={styles.subTitles}>Duration</Text>
-                  <Text style={styles.infobox}>
-                    {selfGuidedItem.duration} hours
-                  </Text>
-                </View>
-              </View>
-              <View style={styles.flexRow}>
-                <Image
-                  style={styles.contactSocialMediaIcon}
-                  source={require('../../assets/images/icFaveShareDefault.png')}
-                />
-                <TouchableOpacity
-                  onPress={async e => {
-                    try {
-                      if (
-                        savedIds.some(savedId => savedId === selfGuidedItem.id)
-                      ) {
-                        removeSaveId(selfGuidedItem.id);
-                      } else {
-                        addSaveId(selfGuidedItem.id);
-                      }
-                    } catch (error) {}
-                  }}>
+                <View style={styles.flexRow}>
                   <Image
-                    style={styles.contactSocialMediaIcon}
-                    source={
-                      savedIds.some(savedId => savedId === selfGuidedItem.id)
-                        ? require('../../assets/images/icFaveCircleActive.png')
-                        : require('../../assets/images/icFaveCircleDefault.png')
-                    }
+                    style={styles.contactSocialMediaIconShare}
+                    source={require('../../assets/images/icFaveShareDefault.png')}
                   />
-                </TouchableOpacity>
+                  <TouchableOpacity
+                    onPress={async e => {
+                      try {
+                        if (
+                          await savedIds.some(
```

Comment:
saveIds is an array. I don't think you need await on any of these methods. Remove asyn await try catch.

Reply:
Removed